base/stream.jl

```diff
@@ -1148,6 +1148,14 @@ unmark(x::LibuvStream)   = unmark(x.buffer)
 reset(x::LibuvStream)    = reset(x.buffer)
 ismarked(x::LibuvStream) = ismarked(x.buffer)
 
+function peek(s::LibuvStream)
```

Comment:
Why is `peek(s::IO)` not sufficient here?

Reply:
`LibuvStream` isn't defined when that method is defined; I don't recall if I tried making the definition here. In general, this IO stuff is a bit of a mess because we have Julia objects, Libuv objects, flisp/support objects, and raw system objects. This all needs to be significantly simplified.